bin/pycbc_condition_strain

```diff
@@ -45,28 +45,32 @@ parser.add_argument('--output-channel-name',
 parser.add_argument('--output-gates-file',
                     help='Save gating info to specified file, in the same '
                          'format as accepted by the --gating-file option')
-parser.add_argument('--single-precision', action='store_true',
-                    help='Output the strain in single precision.')
-parser.add_argument('--no-dyn-range-factor', action='store_true',
-                    help='Do not apply the %f scaling factor to the output '
-                         'strain. Use this to generate frames of simulated '
-                         'data. Incompatible with --single-precision as that '
-                         'would produce underflows.' % pycbc.DYN_RANGE_FAC)
+parser.add_argument('--output-precision', type=str,
+                    choices=['single', 'double'], default='double',
+                    help='Precision of output strain, %(default)s by default')
+parser.add_argument('--dyn-range-factor', action='store_true',
+                    help='Scale the output strain by a large factor (%f) '
+                         'to avoid underflows in subsequent '
+                         'calculations' % pycbc.DYN_RANGE_FAC)
 parser.add_argument('--low-frequency-cutoff', type=float,
                     help='Provide a low-frequency-cutoff for fake strain. '
                          'This is only needed if fake-strain or '
-                         'fake-strain-from-file is used.')
+                         'fake-strain-from-file is used')
 pycbc.strain.insert_strain_option_group(parser)
 args = parser.parse_args()
 
 logging.basicConfig(format='%(asctime)s %(message)s', level=logging.INFO)
 
-if args.single_precision and args.no_dyn_range_factor:
-    parser.error('Refusing to save strain in single precision without the '
-                 'dynamic range factor, as it would lead to underflows')
-precision = 'single' if args.single_precision else 'double'
-dyn_range_factor = 1 if args.no_dyn_range_factor else pycbc.DYN_RANGE_FAC
-out_strain = pycbc.strain.from_cli(args, dyn_range_factor, precision=precision)
+# read and condition strain as pycbc_inspiral would do
+out_strain = pycbc.strain.from_cli(args, dyn_range_fac=pycbc.DYN_RANGE_FAC)
+
+# force strain precision to be as requested
+out_strain = out_strain.astype(
+        float32 if args.output_precision == 'single' else float64)
```

Comment:
To fix the error @a-r-williamson sees, why not just remove the two lines above, and send the precision key-word argument into from_cli?

Reply:
But that was exactly the problem we were having before no?

Reply:
Now I'm confused! What problem are you referring to. This type conversion would be done within the strain module, no?

https://github.com/ligo-cbc/pycbc/blob/master/pycbc/strain.py#L274

Reply:
The precision argument of `from_cli` influences the internal precision of how autogating is done, which I do not like. What I have done now exactly matches what `pycbc_inspiral` does, I think that is the way it should behave. @a-r-williamson 's error can be fixed trivially by moving a block of code, which I just did. I will upload in a moment, after a quick test.

Reply:
Okay, this suggests that we need to do some cleaning of `from_cli`, but that can wait until O2 is properly over.

Reply:
Patch remains approved when you want to merge